Source/CesiumRuntime/Public/Cesium3DTileset.h

```diff
@@ -242,6 +243,15 @@ class CESIUMRUNTIME_API ACesium3DTileset : public AActor,
       meta = (EditCondition = "EnforceCulledScreenSpaceError", ClampMin = 0.0))
   float CulledScreenSpaceError = 64.0;
 
+  /**
+   * Refreshes this tileset, ensuring that all materials and other settings are
+   * applied. It is not usually necessary to invoke this, but when
+   * behind-the-scenes changes are made and not reflected in the tileset, this
+   * function can help.
+   */
+  UFUNCTION(CallInEditor, BlueprintCallable, Category = "Cesium")
+  void RefreshTileset();
```

Comment:
What is "refreshing"? (Cold softdrinks....?!). **Who** is going to call this function, to achieve **what**? Why is this public and `BlueprintCallable`?  

Right now, it does not seem to be called at all. It only calls `DestroyTileset`, and `DestroyTileset` is already called at 16 (!) different places. Many of these calls essentially convey the message "Something changed, we don't know how to handle that, let's start again from scratch" (and these calls could reasonably be changed to a `RefreshTileset` call, even if that was a private method...). 

More broadly: **If** it is known that something can change (like setting some material parameter value), then that change should be tracked and trigger the "smallest possible update" _internally_. This might, initially, well be a `DestroyTileset` call. But mabe it would be possible to improve that later...

    void layerOpacityChangedOrSo() {
        //RefreshTileset(); // remove this
        // do this instead:
        tileset->someMaterial->someLayer->setOpacity(newOpacity);
    }

Reply:
It's not used anywhere because it's meant to be used by the user in the editor or blueprint, although it may be nice to use it internally in place of DestroyTileset (as you recommended). But either way I think it's useful as a catch-all for now. There are a lot of cases where the refresh _may_ be needed, such as adding/removing/editing a CartographicPolygon, editing material layers, editing the blend functions.

Reply:
> If it is known that something can change (like setting some material parameter value), then that change should be tracked and trigger the "smallest possible update" internally.

Yes, of course, whenever possible. But it can get quite complicated. My tileset references a material which uses a material function which uses another material function which uses a texture which I've just re-imported. We might even handle that one, but I'm not sure and in any case there are probably some (rare?) cases that we don't. As the Doxygen says, it's usually not necessary to use RefreshTileset, but it's nice to have the option.

Reply:
It is not clear under which conditions the function should be called. It is not clear what the function does. We will never be able to remove that function, or change its semantics in any way.

People will call this function, because ~"something does not work, and after calling the function, it does work". They will not hesitate to do this, because "refresh" sounds nice and innocent. Consider calling it `CompletelyDestroyAndReInitializeTileset` - because _that's what it does_ (one could argue to just make `DestroyTileset` a `public` function - the fact that the tileset is re-initialized automatically on the fly after it was destroyed is a subtle implementation detail).